dev-docs/RFCs/vNext/fixed-frame-coordinates-rfc.md

```diff
@@ -0,0 +1,115 @@
+
+This puts us in a position where we now have both the necessary support libraries, an initial use case for, and a more mature understanding of fixed frame geospatial coodinates, so it makes sense to create an RFC to dig into what adding fixed frame coordinate support directly to deck.gl would mean.
+
+
+## Overview
+
+To make support for fixed frame coordinates into a "first class citizen" we would ideally added it in several places in the API:
+* Layers: New coordinate system
+* View States: Support fixed frame view states
+* Unprojection/Picking: Returned coordinates
+
+
+## Use Cases
+
+The 3D tiles implementation in loaders.gl would an important pilot use case, making sure that fixed frame support in deck.gl is built for a real, testable use case.
+
+In addition, the `Tile3DLayer` implementation would be slightly simplified as it would be able to rely on deck.gl. The size/complexity wins in loaders.gl are welcome but not that significant, however the debuggability/maintainability of the Tile3DTraversal code might be considerably improved by having a cleaner end-to-end system, which is important as the long-term maintenance of this code needs to be secored.
+
+We are also encountering requests for fixed frame coordinates from potential users of other vis.gl frameworks, e.g. AVS.auto, and being able to say that our stack supports them "ground up" might be important to unblock certain partnerships.
+
+Longer-term, the ability to work in a fixed, uniform space across all deck.gl layers/data could significantly simplify the use of traditional 3D techniques, and also simplify 3D globe type visualizations, e.g. enabling us to integrate more closely with systems such as Cesium.
+
+
+## Proposals
+
+### Proposal: New COORDINATE_SYSTEM.WGS84 for layers
+
+The most important addition would be to support a new `COORDINATE_SYSTEM.FIXED_FRAME` coordinate system in layers enabling users to supply data in fixed frame coordinates.
+
+```js
+new Layer({
+  coordinateSystem: COORDINATE_SYSTEM.FIXED_FRAME,
+  data: [...] // coordinates in WGS84 cartesian
+})
+```
+
+Layers can currently specify coordinates in `COORDINATE_SYSTEM.LNGLAT`, or `COORDINATE_SYSTEM.METER_OFFSETS` (local East-North-Up coordinates from an origin).
+
+Implementation-wise, our layer shaders would use `@math.gl/geospatial` to generate a transform from fixed coordinates to a local ENU coordinate system, multiply that with any model matrix, and then treat the layer data `COORDINATE_SYSTEM.METER_OFFSETS`.
+
+The local ENU coordinate system could have its origin in the current view state center.
+
+Other name ideas for the new coordinate system mode: `COORDINATE_SYSTEM.WGS84`, `COORDINATE_SYSTEM.WGS84_CARTESIAN`, `COORDINATE_SYSTEM.ECF`, ...
```

Comment:
It's worth noting that our current solution to position precision is a `vec2 instancePositions64xyLow` attribute, which is tailored to the LNGLAT use case. To support fixed frame, and other future coordinate systems, we need to start treating z as 64-bit as well. This will require changes to all core layer shaders and the project module.

Reply:
Great point. I will add this to the RFC. 
 
I would certainly argue that those changes should be done regardless. The more we can generalize and standardize our coordinate system handling (without loosing perf) the better off we will be.

But of course it should be called out.

Reply:
This comment should not longer be relevant given the v8.0 coordinate system refactor in #3835